src/main/webapp/js/index.js

```diff
@@ -38,35 +39,31 @@ async function loadLogoutButton(account) {
  */
 function getAllReceipts() {
   const params = new URLSearchParams();
-  params.append('isPageLoad', 'true');
+  isPageLoad = true;
+  params.append('isPageLoad', isPageLoad);
 
   searchReceipts(params);
 }
 
 /** Fetches matching receipts from the server and adds them to the DOM. */
 function getMatchingReceipts() {
   const params = new URLSearchParams();
+  isPageLoad = false;
   params.append('isNewSearch', 'true');
-  params.append('isPageLoad', 'false');
-  params.append('category', document.getElementById('category-input').value);
-  params.append(
-      'dateRange', document.getElementById('date-range-input').textContent);
-  params.append('store', document.getElementById('store-name-input').value);
-  params.append('min', document.getElementById('min-price-input').value);
-  params.append('max', document.getElementById('max-price-input').value);
-  const dateTimeFormat = new Intl.DateTimeFormat();
-  params.append('timeZoneId', dateTimeFormat.resolvedOptions().timeZone);
+  params.append('isPageLoad', isPageLoad);
+  appendQueryInformation(params);
 
   searchReceipts(params);
 }
 
 /** Fetches next receipts page from the server and adds it to the DOM. */
 function getNextPageOfReceipts() {
   const params = new URLSearchParams();
-  params.append('isPageLoad', 'false');
-  params.append('isNewSearch', 'false');
+  params.append('isPageLoad', isPageLoad);
+  params.append('isNewSearch', !isPageLoad);
```

Comment:
Seems like you can just eliminate isNewSearch from the javascript and servlet altogether?